lib/pool.js

```diff
@@ -1,48 +1,77 @@
-var _ = require('lodash'),
-  SandCastle = require("./sandcastle").SandCastle;
+const SandCastle = require("./sandcastle").SandCastle;
+const fs = require('fs');
+const MMAP = require('mmap-object');
 
-function Pool(opts, sandCastleCreationOpts) {
-  _.extend(this, {
+const Pool = function(opts, sandCastleCreationOpts) {
+  Object.assign(this, {
     numberOfInstances: 1
   }, opts);
 
-  if(this.numberOfInstances < 1) {
+  if (this.numberOfInstances < 1) {
     throw "Can't create a pool with zero instances";
   }
 
   this.sandcastles = [];
   this.runQueue = [];
   this.consumerSleeping = true;
 
-  for(var i = 0; i < this.numberOfInstances; ++i) {
-    var sandCastleOptions = {};
-    _.extend( sandCastleOptions,
-              sandCastleCreationOpts,
-              { socket: '/tmp/sandcastle_' + i.toString() + '.sock' }
-            );
+  if (sandCastleCreationOpts) {
+    if (sandCastleCreationOpts.memoryLimitMB) {
+      this.createMmap(sandCastleCreationOpts.memoryLimitMB);
+    } else {
+      this.createMmap(0);
+    }
+  } else {
+    this.createMmap(0);
+  }
+
+  const sandCastleOptions = {};
+  for (let i = 0; i < this.numberOfInstances; ++i) {
+    Object.assign( sandCastleOptions,
+      sandCastleCreationOpts, {
+        socket: `/tmp/sandcastle_${i.toString()}.sock`,
+        sharedFileID: this.ufid,
+        sharedObject: this.mmapObject
+      }
+    );
 
     this.sandcastles.push({
       castle: new SandCastle(sandCastleOptions),
       running: false
     });
   }
-}
+};
+
+Pool.prototype.createMmap = function(memoryLimit) {
```

Comment:
could you share some of this createMmap code between the two spots? (the pool and the individual sandcastle)